core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -130,15 +131,22 @@ private[spark] class BlockManager(
     numUsableCores: Int)
   extends BlockDataManager with BlockEvictionHandler with Logging {
 
-  private[spark] val externalShuffleServiceEnabled =
+  private[spark] val externalNonK8sShuffleService =
     conf.get(config.SHUFFLE_SERVICE_ENABLED)
+
+  private[spark] val externalk8sShuffleServiceEnabled =
+    conf.get(config.K8S_SHUFFLE_SERVICE_ENABLED)
```

Comment:
I think there's some overlap between this config and the `SHUFFLE_SERVICE_PROVIDER_CLASS` config. Perhaps we can use this flag to determine which provider class to load, thus simplifying the configs?